govcd/vdc_group.go

```diff
@@ -0,0 +1,500 @@
+/*
+ * Copyright 2021 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"net/url"
+)
+
+// VdcGroup is a structure defining a VdcGroup in Organization
+type VdcGroup struct {
+	VdcGroup *types.VdcGroup
+	Href     string
+	client   *Client
+	parent   organization
+}
+
+// CreateNsxtVdcGroup create NSX-T VDC group with provided VDC IDs.
+// More generic creation method available also - CreateVdcGroup
+func (adminOrg *AdminOrg) CreateNsxtVdcGroup(name, description, startingVdcId string, participatingVdcIds []string) (*VdcGroup, error) {
+	participatingVdcs, err := constructParticipatingOrgVdcs(adminOrg, startingVdcId, participatingVdcIds)
+	if err != nil {
+		return nil, err
+	}
+
+	vdcGroupConfig := &types.VdcGroup{}
+	vdcGroupConfig.OrgId = adminOrg.orgId()
+	vdcGroupConfig.Name = name
+	vdcGroupConfig.Description = description
+	vdcGroupConfig.ParticipatingOrgVdcs = participatingVdcs
+	vdcGroupConfig.LocalEgress = false
+	vdcGroupConfig.UniversalNetworkingEnabled = false
+	vdcGroupConfig.NetworkProviderType = "NSX_T"
+	vdcGroupConfig.Type = "LOCAL"
+	vdcGroupConfig.ParticipatingOrgVdcs = participatingVdcs
+	return adminOrg.CreateVdcGroup(vdcGroupConfig)
+}
+
+func constructParticipatingOrgVdcs(adminOrg *AdminOrg, startingVdcId string, participatingVdcIds []string) ([]types.ParticipatingOrgVdcs, error) {
```

Comment:
Please add a comment explaining what the function does for a casual reader.
Also, why `constructParticipatingOrgVdcs` and not `createParticipatingOrgVdcs`?

Reply:
Added comment for function.
This is type converter function, there isn't a need to create anything. In summary, fetch and convert.

Reply:
We haven't used `construct` before.

@dataclouder Would you have a better prefix for such functions handy?

Reply:
Just a few other options `build`, `compose`. Although I don't have a very strong opinion against construct

Reply:
I'm OK with `compose` too, as we have it in some other places @vbauzysvmware

Reply:
changed